keps/sig-scheduling/1258-default-pod-topology-spread/README.md

```diff
@@ -259,13 +272,200 @@ To ensure this feature to be rolled out in high quality. Following tests are man
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+
+  Running workloads are not affected by `kube-scheduler`.
+
+* **What specific metrics should inform a rollback?**
+
+  Primarily scheduling latency metrics, such as `framework_extension_point_duration_seconds`, `scheduling_algorithm_duration_seconds`
+  and `e2e_scheduling_duration_seconds`, when they have increased significantly.
+
+  Since spreading is affected, node utilization might change.
+  Utilization metrics can be queried in the `/metrics/resource` endpoint exposed by kubelet.
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+
+  N/A.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
+  fields of API types, flags, etc.?**
+
+  TBD for GA.
+
+### Monitoring Requirements
+
+* **How can an operator determine if the feature is in use by workloads?**
+
+  All Pods are affected, unless they have explicit spreading constraints (.spec.topologySpreadConstraints).
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine 
+  the health of the service?**
+
+  - [x] Metrics
+    - Metric name: `framework_extension_point_duration_seconds` with label `extension_point` values `PreScore` and/or `Score`.
+    - [Optional] Aggregation method:
+    - Components exposing the metric: `kube_scheduler`.
+  - [ ] Other (treat as last resort)
+    - Details:
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+
+  For 100 nodes, with a 4-core master:
+
+  - Latency for PreScore less than 15ms for 99% percentile.
+  - Latency for Score less than 50ms for 99% percentile.
+
+* **Are there any missing metrics that would be useful to have to improve observability 
+of this feature?**
+
+  N/A.
+
+### Dependencies
+
+* **Does this feature depend on any specific services running in the cluster?**
+
+  N/A.
+
+
+### Scalability
+
+* **Will enabling / using this feature result in any new API calls?**
+
+  No.
+
+* **Will enabling / using this feature result in introducing new API types?**
+
+  No.
+
+* **Will enabling / using this feature result in any new calls to the cloud 
+provider?**
+
+  No.
+
+* **Will enabling / using this feature result in increasing size or count of 
+the existing API objects?**
+
+  No.
+
+* **Will enabling / using this feature result in increasing time taken by any 
+operations covered by [existing SLIs/SLOs]?**
+
+  Scheduling time on clusters with more than 100 nodes. Smaller clusters are unaffected.
+  `SelectorSpreading` doesn't take into account all the Nodes in big clusters when calculating skew,
+  resulting in partial spreading at this scale.
+  On the contrary, `PodTopologySpreading` considers all nodes when using topologies bigger than a Node, like a Zone.
```

Comment:
Do we have any numbers?

Reply:
In synthetic unit level benchmarks, difference is negligible. But we will have more precise numbers from dashboards when we enable the feature by default (beta).

Reply:
Re benchmarks: even in 5k clusters? Are those using the same config we will have in real clusters?

Reply:
The benchmark is actually a bit outdated, but it's CPU footprint is the same. And the benchmark is for 1000 nodes.
 https://github.com/kubernetes/kubernetes/blob/6e3ef0be163566c08398e1e5ff43f87accfb036b/pkg/scheduler/framework/plugins/podtopologyspread/scoring_test.go#L754

This is the legacy counterpart https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/selectorspread/selector_spread_perf_test.go

Reply:
OK - that's fine.

Can you please add a note to ensure to validate it during graduation?
Something like:
"Before graduation we will ensure that the latency increase is acceptable with Scalability SIG"
(or sth along those lines).

Reply:
Done